spec/models/user_spec.rb

```diff
@@ -1,5 +1,45 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+
+  describe 'validations' do
+    it { should validate_presence_of(:unit) }
+    it { should validate_presence_of(:role) }
+    it { should validate_presence_of(:first_name) }
+    it { should validate_presence_of(:last_name) }
+    it { should validate_presence_of(:building_id) }
+    it { should validate_presence_of(:neighborhood_id) }
+
+    it { should have_valid(:first_name).when('John') }
+    it { should_not have_valid(:first_name).when(nil, '') }
+
+    it { should have_valid(:last_name).when('Smith') }
+    it { should_not have_valid(:last_name).when(nil, '') }
+
+    it { should have_valid(:email).when('john@example.com') }
+    it { should_not have_valid(:email).when(
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.